include/zephyr/drivers/interrupt_controller/riscv_plic.h

```diff
@@ -12,27 +12,77 @@
 #ifndef ZEPHYR_INCLUDE_DRIVERS_RISCV_PLIC_H_
 #define ZEPHYR_INCLUDE_DRIVERS_RISCV_PLIC_H_
 
+#include <zephyr/device.h>
+
+/**
+ * @brief Install the ISR into the table
+ *
+ * @param irq Interrupt ID
+ * @param routine Pointer to the ISR handler
+ * @param param Pointer to the argument to be passed into the ISR handler
+ */
+void z_riscv_plic_isr_install(unsigned int irq, unsigned int priority,
+			      void (*routine)(const void *), const void *param);
+
+/**
+ * @brief Install the ISR of an interrupt controller into the table
+ * 
+ * See @ref z_riscv_plic_isr_install for more info
+ *
+ * @param intc Interrupt controller of the irq, the first instance will be used if this is NULL
+ */
+void z_riscv_plic_isr_intc_install(const struct device *intc, unsigned int irq,
+				   unsigned int priority, void (*routine)(const void *),
+				   const void *param);
+
 /**
  * @brief Enable interrupt
  *
  * @param irq interrupt ID
  */
-void riscv_plic_irq_enable(uint32_t irq);
+void riscv_plic_irq_enable(uint32_t irq, const struct device *dev);
```

Comment:
The signature for this should probably not be changed.

Reply:
In this approach, the IRQN doesn't carry information about the plic instance, so the `dev` is required